File: gradle/packages/cli-function/src/main/kotlin/com/app/cli/function/Number.kt
package com.app.cli.function

import com.app.numbers.utils.factorial
import com.app.numbers.utils.fibonacci

object Number {
    @JvmStatic
    fun main(args: Array<String>) {
        println("===================================================")
        printNumberCalculations()
        println("===================================================")
    }

    private fun getUserNumber(): Int {
        print("Enter a number : ")
        return readLine()?.trim()?.toIntOrNull() ?: 0
    }

    fun printNumberCalculations() {
        val number = getUserNumber()
        println("Fib[$number] = ${fibonacci(number)}")
        println("Fact[$number] = ${factorial(number)}")
    }
}

File: dotnet/src/CLI.Function/Number/Number.cs
using Mathematics = Prog.Numbers.Utils.Operations;

namespace Prog.CLI.Function;

public class Number
{
    public static int GetUserNumber()
    {
        Console.Write("Enter a number : ");
        string numberStr = Console.ReadLine()?.Trim() ?? string.Empty;
        return int.TryParse(numberStr, out int number) ? number : 0;
    }

    public static void PrintNumberCalculations()
    {
        int number = GetUserNumber();
        Console.WriteLine($"Fib[{number}] = {Mathematics.FibonacciUtils.Fibonacci(number)}");
        Console.WriteLine($"Fact[{number}] = {Mathematics.FactorialUtils.Factorial(number)}");
    }
}
